bodo/pandas/frame.py

```diff
@@ -38,7 +38,19 @@ def _plan(self):
             if self._mgr._plan is not None:
                 return self._mgr._plan
             else:
-                return plan_optimizer.LogicalGetDataframeRead(self._mgr._md_result_id)
+                if hasattr(self, "source_plan") and self.source_plan is not None:
+                    return self.source_plan
+                from bodo.pandas.base import _empty_like
+
+                out_schema = _empty_like(self)
+                if bodo.dataframe_library_run_parallel:
+                    res_id = bodo.spawn.utils.scatter_data(self)
```

Comment:
Reuse the result id if the data is already distributed.

Reply:
Add a test for the case that the data is not collected to spawner (output of dataframe library to JIT).

Reply:
Done.